src/core/main.c

```diff
@@ -1871,6 +1871,9 @@ static int do_reexecute(
          * SIGCHLD for them after deserializing. */
         if (IN_SET(objective, MANAGER_SWITCH_ROOT, MANAGER_SOFT_REBOOT))
                 broadcast_signal(SIGTERM, /* wait_for_exit= */ false, /* send_sighup= */ true, arg_default_timeout_stop_usec);
+        /* On soft reboot really make sure nothing is left */
+        if (objective == MANAGER_SOFT_REBOOT)
+                broadcast_signal(SIGKILL, /* wait_for_exit= */ false, /* send_sighup= */ false, arg_default_timeout_stop_usec);
```

Comment:
I think we should consider setting `wait_for_exit` to `True` here, so that we wait until all processes have been SIGKILLed, before doing the soft reboot

Reply:
mmh but that's not useful for sigkill, no? For sigterm it's processed by the process so it can take time, but sigkill cannot be blocked - this matches what the shutdown binary does, it waits for term but doesn't for kill

Reply:
Hmm, it shouldn't hurt to add it though since it's instant? I figure for shutdown it doesn't really matter since we're going down anyway, but here any process that for some reason doesn't shut down immediately even on SIGKILL might run in the new rootfs for some amount of time. Seems safer to just wait for them to shut down (especially since there's a timeout on the operation regardless)

Reply:
It's not waiting on term either though right now, if you check the existing bit, the idea is that processed will be reaped by the restarted instance. I don't think it's about safety, it's just about zombie processes

Reply:
For `sigterm` it makes sense as those can be ignored by the process whereas SIGKILL cannot be ignored, so if the wait isn't instant for SIGKILL that'd be a special case scenario.

Anyway, it's a bit of a wash, probably doesn't matter that much in the end.

Reply:
if we see that there are zombie processes left after the reboot we can revisit